### PR TITLE
*: support display tiflash ru in explain analyze

### LINKED ISSUE #50637: support display ru in explain analyze and statments log

### PR DESCRIPTION
**Pull request by guo-shaoge**
### What problem does this PR solve?

Issue Number: close #50637

Problem Summary: when got execution summary, try merge it into `ctx.Value(clientutil.RUDetailsCtxKey)`, which will be displayed in explain analyze and statsment log.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

tpch50 Q9
image

image

Query with limit(https://github.com/pingcap/tidb/issues/50493): 
cf0425d4-9374-4f2d-8363-69f0fab5dc5a

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
support display tiflash ru usage in explain analyze
```

### CODE REVIEW

**Review by AilinKid — APPROVED** [review]

**Review by wshwsh12 — APPROVED** [review]

File: pkg/executor/explain.go [hunk 1]
```diff
@@ -105,6 +124,10 @@ func (e *ExplainExec) executeAnalyzeExec(ctx context.Context) (err error) {
 					err = err1
 				}
 			}
+
+			// Handle RU runtime stats again after Close() to make sure all ru has been collected.
```
Comment (windtalker) [inline pkg/executor/explain.go:128]:
```suggestion
			// Handle RU runtime stats after Close() to make sure all ru has been collected.
```

**Review by windtalker — APPROVED** [review]
LGTM

File: pkg/executor/explain.go (hunk 1 above)
Comment (guo-shaoge) [reply pkg/executor/explain.go:128]: done

**Review by zeminzhou — APPROVED** [review]

**Pull request merged**